### PR TITLE
Deprecate MutArray/Stream module, move funcs to MutArray module

### PR DESCRIPTION
**Pull request by harendra-kumar**
Move the mut array compaction funcs from MutArray.Stream to MutArray module.

### CODE REVIEW

File: core/src/Streamly/Internal/Data/MutArray/Type.hs [hunk 1]
```diff
@@ -2582,7 +2596,240 @@ byteEq :: MonadIO m => MutArray a -> MutArray a -> m Bool
...
--- NFData
+-- Compact
+-------------------------------------------------------------------------------
+
+-- | Parser @pCompactChunksLE n@ coalesces adjacent arrays in the input stream
+-- only if the combined size would be less than or equal to n.
+--
+-- /Internal/
+{-# INLINE_NORMAL pCompactChunksLE #-}
+pCompactChunksLE ::
```
Comment (adithyaov) [inline core/src/Streamly/Internal/Data/MutArray/Type.hs:2607]: Does `p` here signify parser?

File: core/src/Streamly/Internal/Data/MutArray/Type.hs [hunk 2]
```diff
@@ -2582,7 +2596,240 @@ byteEq :: MonadIO m => MutArray a -> MutArray a -> m Bool
...
+    step' _ (SpliceYielding arr next) = return $ D.Yield arr next
+
+-- | Fold @fCompactChunksGE n@ coalesces adjacent arrays in the input stream
+-- until the size becomes greater than or equal to n.
+--
+{-# INLINE_NORMAL fCompactChunksGE #-}
+fCompactChunksGE ::
+       forall m a. (MonadIO m, Unbox a)
+    => Int -> FL.Fold m (MutArray a) (MutArray a)
+fCompactChunksGE n = Fold step initial extract extract
```
Comment (adithyaov) [inline core/src/Streamly/Internal/Data/MutArray/Type.hs:2710]: The name `Chunks` is already here. Can we move this to the fold module?

File: core/src/Streamly/Internal/Data/MutArray/Type.hs [hunk 3]
```diff
@@ -2582,7 +2596,240 @@ byteEq :: MonadIO m => MutArray a -> MutArray a -> m Bool
...
+
+    extract Nothing = return nil
+    extract (Just buf) = return buf
+
+-- | Like 'compactChunksGE' but for transforming folds instead of stream.
+--
+-- >>> lCompactChunksGE n = Fold.many (MutArray.fCompactChunksGE n)
+--
+{-# INLINE_NORMAL lCompactChunksGE #-}
+lCompactChunksGE :: (MonadIO m, Unbox a)
```
Comment (adithyaov) [inline core/src/Streamly/Internal/Data/MutArray/Type.hs:2748]: Can we move this to the fold module?

File: core/src/Streamly/Internal/Data/MutArray/Type.hs (hunk 1 above)
Comment (adithyaov) [reply core/src/Streamly/Internal/Data/MutArray/Type.hs:2607]: Can we move this to the parser module? `Chunks` is already in the name.

Comment (adithyaov) [reply core/src/Streamly/Internal/Data/MutArray/Type.hs:2607]: If we want to keep it here, can we rename it to `compactChunksParserLE`? Although long it's explicit.
The prefix `p` is confusing.

File: core/src/Streamly/Internal/Data/MutArray/Type.hs (hunk 2 above)
Comment (adithyaov) [reply core/src/Streamly/Internal/Data/MutArray/Type.hs:2710]: Rename it as it was renamed earlier. The prefix `f` is confusing.

File: core/src/Streamly/Internal/Data/MutArray/Type.hs [hunk 4]
```diff
@@ -2582,7 +2596,240 @@ byteEq :: MonadIO m => MutArray a -> MutArray a -> m Bool
...
+-- instead of mutating it.
+
+-- | Scan @rCompactChunksLE n@ coalesces adjacent arrays in the input stream
+-- only if the combined size would be less than or equal to n.
+--
+-- /Internal/
+{-# INLINE_NORMAL rCompactChunksLE #-}
+rCompactChunksLE :: (MonadIO m, Unbox a)
+    => Int -> D.Stream m (MutArray a) -> D.Stream m (MutArray a)
+rCompactChunksLE n (D.Stream step state) =
```
[the remark this answers is not in the document]
Comment (adithyaov) [reply core/src/Streamly/Internal/Data/MutArray/Type.hs:2662]: Why do you call this a `Scan`?

File: core/src/Streamly/Internal/Data/MutArray/Type.hs [hunk 5]
```diff
@@ -2582,7 +2596,240 @@ byteEq :: MonadIO m => MutArray a -> MutArray a -> m Bool
...
+    | SpliceBuffering s arr
+    | SpliceYielding arr (SpliceState s arr)
+    | SpliceFinish
+
+-- This mutates the first array (if it has space) to append values from the
+-- second one. This would work for immutable arrays as well because an
+-- immutable array never has additional space so a new array is allocated
+-- instead of mutating it.
+
+-- | Scan @rCompactChunksLE n@ coalesces adjacent arrays in the input stream
```
Comment (adithyaov) [inline core/src/Streamly/Internal/Data/MutArray/Type.hs:2655]: `Scan`?

File: core/src/Streamly/Internal/Data/MutArray/Type.hs (hunk 1 above)
Comment (harendra-kumar) [reply core/src/Streamly/Internal/Data/MutArray/Type.hs:2607]: it's been renamed to pCompactLE in a later PR. adding Parser/Fold suffix may be verbose especially if we follow this convention in more places.

File: core/src/Streamly/Internal/Data/MutArray/Type.hs (hunk 4 above)
[the remark this answers is not in the document]
Comment (harendra-kumar) [reply core/src/Streamly/Internal/Data/MutArray/Type.hs:2662]: r => right, l => left (e.g. rMapM and lmap in fold module), used when we need to differentiate between covariant and contravariant style mapping.

In general, I use the term scan for any stateful transformation.

File: core/src/Streamly/Internal/Data/MutArray/Type.hs (hunk 3 above)
Comment (harendra-kumar) [reply core/src/Streamly/Internal/Data/MutArray/Type.hs:2748]: No. this is specific to an array type (e.g. Array, Array.Generic, MutArray, MutArray.Generic etc), you cannot have all these functions in Fold module. Also Fold module should not have functions for all types, the module for MutArray type should have folds operating on this type.

File: core/src/Streamly/Internal/Data/MutArray/Type.hs (hunk 5 above)
Comment (harendra-kumar) [reply core/src/Streamly/Internal/Data/MutArray/Type.hs:2655]: stateful transformation.

**Pull request merged**